docs/spec_draft.md

```diff
@@ -1798,6 +1799,92 @@ tensor and produces a `result` tensor.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.custom_call
+
+### Semantics
+
+Invokes exteral code with `inputs` as arguments and produces `results`. The
+exact mechanism is backend-specific. The inputs are described as follows:
+  * `inputs` are passed as agruments to the external API.
+  * `call_target_name` may be used as an identifier for the external code.
+  * `has_side_effect` is true if the custom call has side-effects.
+  * `backend_config` may be used to encode additional backend-specific
+    information.
+  * `api_version` specifies the version of the custom call API.
+  * `called_computations` used to apply functions within the scope of the parent
+    module.
+  * `operand_layouts` specifies the layout of the operand.
+  * `result_layouts` specifies the layout of the result.
+  * `output_operand_aliases` captures the alias relationship of `results` to one of the operands.
+    * `operand_index` denotes the index of the operand `inputs` that relates to `results`.
+    * `operand_tuple_indices` are used to index into the operand at `operand_index`, if it is a tuple/nested tuple.
+    * `output_tuple_indices` are used to index into `results`, if it is a tuple/nested tuple.
+
+### Inputs
+
+| Name                    | Type                                                                |
+|-------------------------|---------------------------------------------------------------------|
+| `inputs`                | variadic number of tensors of any supported type, tokens, or tuples |
+| `call_target_name`      | constant of type `string`                                           |
+| `has_side_effect`       | constant of type `i1`                                               |
+| `backend_config`        | constant of type `string`                                           |
+| `api_version`           | enum of `API_VERSION_ORIGINAL`, `API_VERSION_STATUS_RETURNING`,     |
+|                         | and `API_VERSION_STATUS_RETURNING_UNIFIED`                          |
+| `called_computations`   | variadic number of `function` references                            |
+| `operand_layouts`       | variadic number of 1-dimensional tensor constants of type `si64`    |
+| `result_layouts`        | variadic number of 1-dimensional tensor constants of type `si64`    |
+| `output_tuple_indices`  | 1-dimensional tensor constant of type `si64`                        |
+| `operand_index`         | constant of type `si64`                                             |
+| `operand_tuple_indices` | 1-dimensional tensor constant of type `si64`                        |
```

Comment:
Similarly, let's drop `output_operand_aliases` from the spec for now and leave figuring it out to #740.